apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/ConfigServerEurekaClientConfigure.java

```diff
@@ -0,0 +1,45 @@
+package com.ctrip.framework.apollo.configservice;
+
+import com.ctrip.framework.apollo.biz.eureka.ApolloEurekaClientConfig;
+import com.netflix.appinfo.InstanceInfo;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.context.scope.refresh.RefreshScope;
+import org.springframework.cloud.netflix.eureka.server.event.EurekaInstanceRegisteredEvent;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author : kl
+ * After startup, set FetchRegistry to true, refresh eureka client
+ **/
+@Configuration
+@ConditionalOnProperty(value = "eureka.client.enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
I think this feature should only be activated when apollo is also an eureka server? e.g. when we are using a standalone eureka server, no EurekaInstanceRegisteredEvent will be received

```suggestion
@ConditionalOnProperty(value = {"eureka.client.enabled", "apollo.eureka.server.enabled"}, havingValue = "true", matchIfMissing = true)
```

Reply:
yes